intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -86,30 +88,101 @@
     "IPADDR = str(output.decode()).split()[0]\n",
     "\n",
     "cluster = LocalCUDACluster(ip=IPADDR)\n",
+    "#client = Client(address=\"10.136.7.103:8786\")\n",
     "client = Client(cluster)\n",
     "client"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Use `pynvml` to determine the GPU memory and decide `start_year`, `end_year` and `part_count`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FUNCTION ASSUMES ALL GPUS ARE IDENTICAL WITHIN A NODE\n",
+    "def memory_info():\n",
+    "    pynvml.nvmlInit()\n",
+    "    handle = pynvml.nvmlDeviceGetHandleByIndex(0)\n",
+    "    gpu_mem = pynvml.nvmlDeviceGetMemoryInfo(handle).total\n",
+    "    pynvml.nvmlShutdown()\n",
+    "    return (gpu_mem)\n",
+    "\n",
+    "res = client.run(memory_info)\n",
+    "total_mem = sum(res.values())\n",
+    "min_mem = min(res.values())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ETL memory usage depends on GPU memory per card (not number of GPU's). \n",
+    "# Assumption is 1gb split variant of the data is used\n",
+    "\n",
+    "start_year = None # start year for etl proessing\n",
+    "end_year = None # end year for etl processing (inclusive)\n",
+    "part_count = None # no. of data files to train against\n",
+    "if total_mem >= (512<<30):\n",
```

Comment:
Sorry to throw a wrench in the works, but this looks ugly.

Perhaps we should put this logic in a util script, as its main use is to make automated benchmarking of this workflow easier.

From an "example RAPIDS notebook" perspective, this logic shouldn't be public facing as it's not representative of the way people would use it.

Ideally, with the ongoing dask/cudf changes, this logic would become obsolete and `repartition` will avoid the need to process specific file sizes dependent on GPU type.

Reply:
I agree with you. The logic can easily be moved to a util script. Will move it to another script